### PR TITLE
Dogfood checksum verification

### PR DESCRIPTION
**Pull request by melix**
### Context

This PR introduces checksum verification to the `gradle/gradle` build itself. Checksum verification is not documented at this stage and will only become an official Gradle feature in 6.2. In the meantime we want to experience the pain and possibly discover bugs.

Because we had the `DistributionIntegritySpec` checking the `SHA-256` checksums of dependencies, I took the path of generating the checksums file with:

```
./gradlew --write-verification-metadata sha256
```

However, we can use whatever checksums are provided by the actual library authors (but we should avoid `md5` and `sha1` exclusively, a warning will be issued if that's the case).

I should mention that I didn't verify any of the checksums against publicly available information. However, I did check that _every_ checksum in the original test exists in the metadata file.

### CODE REVIEW

**Comment by melix** [comment]
cc @JLLeitschuh for ref

File: build.gradle.kts [hunk 1]
```diff
@@ -33,6 +33,17 @@ plugins {
     id("org.gradle.ci.tag-single-build") version("0.74")
 }
 
+buildscript {
+    dependencies {
+        constraints {
+            classpath("xerces:xercesImpl:2.12.0") {
+                // it's unclear why we don't get this version directly from buildSrc constraints
+                because("Maven Central and JCenter disagree on version 2.9.1 metadata")
```
Comment (JLLeitschuh) [inline build.gradle.kts]: Have you reported this to the maintainers?

**Comment by JLLeitschuh** [comment]
1. Where is this verification feature actually enabled so it can be used by our CI agents and Gradle developers? I don't see a change to `settings.gradle.kts` for example.
2. What about verifying the dependencies for `buildSrc` and `buildscript` is this also handled by this? If not, what configurations are covered and why have we chosen not to include all configurations?

**Comment by melix** [comment]
>     1. Where is this verification feature actually enabled so it can be used by our CI agents and Gradle developers? I don't see a change to `settings.gradle.kts` for example.
> 

The feature is automatically enabled with the presence of the verification file. No configuration.

>     2. What about verifying the dependencies for `buildSrc` and `buildscript` is this also handled by this? If not, what configurations are covered and why have we chosen not to include all configurations?

It's done too.

Note that the file format will change, this PR is WIP and depends on the state of `master`. It already made us discover some edge cases/things a bit painful to do.

**Pull request merged**